src/densities.jl

```diff
@@ -60,9 +60,10 @@ is not collinear the spin density is `nothing`.
     @assert n_k > 0
 
     # Allocate an accumulator for ρ in each thread for each spin component
-    ρaccus = [similar(view(ψ[1], :, 1), (basis.fft_size..., n_spin))
+    # TODO use something like T = promote_type(Array{eltype(basis)}, eltype(ψ), ...)
```

Comment:
I think the pattern would be `T = promote_type(T, eltype(\psi))` and then just use `T`

Reply:
https://github.com/JuliaMolSim/DFTK.jl/pull/476/commits/6a821bd79a920dafe7b2001db6609d8aba9644c3